x-pack/plugins/observability_solution/observability_ai_assistant_app/public/hooks/use_previous_prompts.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+
+const AI_ASSISTANT_LAST_USED_PROMPT_STORAGE = 'ai-assistant-last-used-prompts';
+
+export function usePreviousPrompts() {
+  const [previousPrompts = [], setPreviousPrompts] = useLocalStorage(
+    AI_ASSISTANT_LAST_USED_PROMPT_STORAGE,
+    [] as string[]
+  );
+
+  return {
+    previousPrompts,
+    addPrompt: (prompt: string) => {
+      if (previousPrompts[0] !== prompt) {
+        // Keep track of the last 5 prompts
+        setPreviousPrompts([prompt].concat(previousPrompts.filter((_, index) => index < 4)));
```

Comment:
what about something like:
```ts
setPreviousPrompt(previous => uniq(previous.concat(prompt).slice(0,5))
```

Reply:
I've tried the suggestion but it doesn't yield the same results. After entering the following prompts in a conversation: 1, 2, 3, 4, 5:

<img width="499" alt="Screenshot 2024-04-05 at 16 13 32" src="https://github.com/elastic/kibana/assets/535564/bb9ef489-f6f8-4d03-87f6-9633e33a5138">

Original implementation:

<img width="529" alt="Screenshot 2024-04-05 at 16 15 11" src="https://github.com/elastic/kibana/assets/535564/848b24cf-a13b-444e-b28a-68d77e9f27a5">